src/zjs_ipm.h

```diff
@@ -26,27 +38,33 @@ enum {
 #define ERROR_IPM_NOT_SUPPORTED                            0x0001
 #define ERROR_IPM_INVALID_PARAMETER                        0x0002
 #define ERROR_IPM_OPERATION_FAILED                         0x0003
+#define ERROR_IPM_OPERATION_NOT_ALLOWED                    0x0004
 
 // Message Types
 
 // AIO
+#ifdef BUILD_MODULE_AIO
 #define TYPE_AIO_OPEN                                      0x0000
 #define TYPE_AIO_PIN_READ                                  0x0001
 #define TYPE_AIO_PIN_ABORT                                 0x0002
 #define TYPE_AIO_PIN_CLOSE                                 0x0003
 #define TYPE_AIO_PIN_SUBSCRIBE                             0x0004
 #define TYPE_AIO_PIN_UNSUBSCRIBE                           0x0005
 #define TYPE_AIO_PIN_EVENT_VALUE_CHANGE                    0x0006
+#endif
```

Comment:
Defining macros like this that are never used (if the corresponding module isn't built) doesn't really hurt anything, so I doubt we need these #ifdefs?